src/meta/src/manager/env.rs

```diff
@@ -35,26 +37,71 @@ use crate::manager::{
     NotificationManagerRef,
 };
 use crate::model::ClusterId;
-use crate::storage::MetaStoreRef;
-#[cfg(any(test, feature = "test"))]
-use crate::storage::{MemStore, MetaStoreBoxExt};
+use crate::storage::{MemStore, MetaStore, MetaStoreBoxExt, MetaStoreRef};
 use crate::MetaResult;
 
+#[derive(Clone)]
+pub enum IdGenManagerImpl {
+    Kv(IdGeneratorManagerRef),
+    Sql(SqlIdGeneratorManagerRef),
+}
+
+impl IdGenManagerImpl {
+    fn assert_kv_ref(&self) -> &IdGeneratorManagerRef {
+        match self {
+            IdGenManagerImpl::Kv(mgr) => mgr,
+            _ => panic!("expect kv id generator manager"),
+        }
+    }
+}
+
+#[derive(Clone)]
+pub enum MetaStoreImpl {
+    Kv(MetaStoreRef),
+    Sql(SqlMetaStore),
+}
+
+impl MetaStoreImpl {
+    fn assert_kv_ref(&self) -> &MetaStoreRef {
```

Comment:
```suggestion
    fn as_kv(&self) -> &MetaStoreRef {
```

Reply:
Maybe "assume_xxx" as it may panic if it is not "kv"? Like https://doc.rust-lang.org/std/mem/union.MaybeUninit.html#method.assume_init